### PR TITLE
display real courses

### PR DESCRIPTION
**Pull request by smilesthesad**
## Implementation Description
Load in real courses as opposed to dummy data on the courses view page.

## Screenshots
Visual depiction of PR
image

## What should reviewers focus on?
- Make sure that real courses are now being displayed at the root URL (localhost:3000)

## Testing Procedure
- Create some courses, note the title and the description
- Observe that when you go to the root URL this is what is being displayed

## Checklist
- [X] Ensure code follows style guide by running the linters
- [X] I have updated the documentation or deemed it unnecessary
- [X] I have linked the relevant issue in this PR
- [X] I have requested a review from the PL, as well as other dev(s) (and designers if necessary)

### CODE REVIEW

File: frontend/src/APIClients/BaseAPIClient.ts [hunk 1]
```diff
@@ -73,9 +73,13 @@ const accessTokenInjectionLink = new ApolloLink(
...
   (operation) =>
-    ["Refresh", "ResetPassword", "Login", "Signup_Register"].includes(
-      operation.operationName,
-    ),
+    [
+      "Refresh",
+      "ResetPassword",
+      "Login",
+      "Signup_Register",
+      "Courses",
```
Comment (jfdoming) [inline frontend/src/APIClients/BaseAPIClient.ts:81]: Can you double check that the module editor still works with a stale account?

File: frontend/src/components/learner/browser/CourseCard.tsx [hunk 2]
```diff
@@ -63,13 +77,13 @@ const CourseCard = (): React.ReactElement => {
               "-webkit-box-orient": "vertical",
             }}
           >
-            Description description description description description
-            description description
+            {hidden ? "Description..." : course.description}
```
Comment (jfdoming) [inline frontend/src/components/learner/browser/CourseCard.tsx:80]: How about "No description provided"? That way, it doesn't seem like an accident.

File: frontend/src/components/pages/Default.tsx [hunk 3]
```diff
@@ -68,8 +35,12 @@ const Default = (): React.ReactElement => {
           width="100%"
           spacing={5}
         >
-          {courses?.map((course) => (
-            <CourseCard key={course.id} />
+          {courses?.map((course: CourseResponse) => (
+            <CourseCard
+              key={course.id}
+              course={course}
+              hidden={!authenticatedUser}
```
Comment (jfdoming) [inline frontend/src/components/pages/Default.tsx:42]: Could you remove this functionality? We want anyone to be able to browse courses, eg to pique their interest.

File: frontend/src/components/learner/browser/CourseCard.tsx [hunk 4]
```diff
@@ -63,13 +77,13 @@ const CourseCard = (): React.ReactElement => {
...
           </Box>
           <Spacer />
           <Box>
             <Icon as={DocumentIcon} marginTop="-5px" marginRight="5px" />
-            insert course info
+            {hidden ? "N" : moduleCount} Modules • {hidden ? "N" : lessonCount}{" "}
```
Comment (jfdoming) [inline frontend/src/components/learner/browser/CourseCard.tsx:85]: Would suggest to use https://github.com/uwblueprint/rowan-house/blob/main/frontend/src/components/learner/ModuleLessonCount.tsx instead

**Review by jfdoming — CHANGES_REQUESTED** [review]

File: frontend/src/APIClients/BaseAPIClient.ts (hunk 1 above)
Comment (smilesthesad) [reply frontend/src/APIClients/BaseAPIClient.ts:81]: image
nope

**Review by jfdoming — APPROVED** [review]

**Review by a4bello — COMMENTED** [review]
Extensively test and it works well! Great job Victor!

**Review by a4bello — APPROVED** [review]

**Comment by draymond63** [comment]
This is great! But, clicking through still doesn't work right? The router link on the course card still goes to /123 @jfdoming

**Comment by draymond63** [comment]
Oh I can just do it in my PR if that works. It's a one-line change

**Comment by jfdoming** [comment]
Sounds good

**Pull request merged**